dashboard/app/assets/javascripts/header.js.erb

```diff
@@ -413,3 +416,47 @@ dashboard.header.updateTimestamp = function () {
     $('.project_updated_at').text("Not saved"); // TODO i18n
   }
 };
+
+function getSummarizedProgressForAnonymousUser () {
+  var summarizedProgress = {};
+  var levelProgress = {};
+  var cookie = document.cookie.split(';');
```

Comment:
I think there's probably value in finding a good library to do cookie parsing now that we can use going forward. I don't personally have a lot of experience with cookie libraries. One that Phil referenced/found on google that looks both small and promising is https://github.com/ScottHamper/Cookies